src/Analyzers/Core/Analyzers/SimplifyInterpolation/Helpers.cs

```diff
@@ -92,29 +99,26 @@ private static IOperation Unwrap(IOperation expression)
                 !syntaxFacts.IsBaseExpression(invocation.Instance!.Syntax) &&
                 !invocation.Instance.Type!.IsRefLikeType)
             {
-                if (invocation.Arguments.Length == 1 &&
-                    invocation.Arguments[0].Value is ILiteralOperation { ConstantValue: { HasValue: true, Value: string value } } literal &&
-                    invocation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.IFormattable).FullName!) is { } systemIFormattable &&
-                    invocation.Instance.Type.Implements(systemIFormattable))
-                {
-                    unwrapped = invocation.Instance;
-                    formatString = value;
-
-                    var unwrappedSyntax = GetPreservedInterpolationExpressionSyntax<TConditionalExpressionSyntax, TParenthesizedExpressionSyntax>(unwrapped);
-                    unnecessarySpans.AddRange(invocation.Syntax.Span
-                        .Subtract(unwrappedSyntax.FullSpan)
-                        .Subtract(GetSpanWithinLiteralQuotes(virtualCharService, literal.Syntax.GetFirstToken())));
-                    return;
-                }
-
-                var method = invocation.TargetMethod;
-                while (method.OverriddenMethod != null)
+                if (invocation.Arguments.Length == 1
+                    || (invocation.Arguments.Length == 2 && IsInvariantCultureReference(invocation.Arguments[1].Value) && IsInsideFormattableStringInvariant(invocation)))
                 {
-                    method = method.OverriddenMethod;
+                    if (invocation.Arguments[0].Value is ILiteralOperation { ConstantValue: { HasValue: true, Value: string value } } literal &&
+                       invocation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.IFormattable).FullName!) is { } systemIFormattable &&
+                       invocation.Instance.Type.Implements(systemIFormattable))
+                    {
+                        unwrapped = invocation.Instance;
+                        formatString = value;
+
+                        var unwrappedSyntax = GetPreservedInterpolationExpressionSyntax<TConditionalExpressionSyntax, TParenthesizedExpressionSyntax>(unwrapped);
+                        unnecessarySpans.AddRange(invocation.Syntax.Span
+                            .Subtract(unwrappedSyntax.FullSpan)
+                            .Subtract(GetSpanWithinLiteralQuotes(virtualCharService, literal.Syntax.GetFirstToken())));
+                        return;
+                    }
                 }
 
-                if (method.ContainingType.SpecialType == SpecialType.System_Object &&
-                    method.Name == nameof(ToString))
+                if (IsObjectToStringOverride(invocation.TargetMethod)
+                    || (invocation.Arguments.Length == 1 && IsInvariantCultureReference(invocation.Arguments[0].Value) && IsInsideFormattableStringInvariant(invocation)))
```

Comment:
can you extract the `IsInvariantCultureReference(invocation.Arguments[0].Value) && IsInsideFormattableStringInvariant(invocation)` piece.  it's almost identical above (just the different arg/index).